internal/virtualcircuit/update.go

```diff
@@ -0,0 +1,136 @@
+package virtualcircuit
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	metal "github.com/equinix-labs/metal-go/metal/v1"
+	"github.com/spf13/cobra"
+)
+
+type vcUpdateParamBuilder interface {
+	SetName(string)
+	SetDescription(string)
+	SetTags([]string)
+}
+
+// compile-time assertion that parameter types implement vcParamBuilder
+var (
+	_ vcParamBuilder = (*metal.VlanVirtualCircuitCreateInput)(nil)
+	_ vcParamBuilder = (*metal.VrfVirtualCircuitCreateInput)(nil)
+	_ vcParamBuilder = (*metal.VlanVirtualCircuitUpdateInput)(nil)
+	_ vcParamBuilder = (*metal.VrfVirtualCircuitUpdateInput)(nil)
+)
+
+func updateParams(p vcUpdateParamBuilder, name, description string, tags []string) {
+	if name > "" {
+		p.SetName(name)
+	}
+	if description > "" {
+		p.SetDescription(description)
+	}
+
+	if len(tags) > 0 {
+		p.SetTags(tags)
+	}
+}
+
+func (c *Client) Update() *cobra.Command {
+	var (
+		description string
+		speed       string
```

Comment:
Same point here, with taking so many flags as input. It will get very difficult to maintain. Lets think of a config file

Reply:
Same response applies here: https://github.com/equinix/metal-cli/pull/369/files#r1440938204